File: 495-teemo-attacking/teemo-attacking.cs
public class Solution {
    public int FindPoisonedDuration(int[] timeSeries, int duration) {
      int output = 0;
      for(int i = 0; i < timeSeries.Length-1;i++){
        output += Math.Min(duration, timeSeries[i+1]-timeSeries[i]);
      }
      return output + duration;  
    }
}

File: 495-teemo-attacking/teemo-attacking.js
var findPoisonedDuration = function(timeSeries, duration) {
    let curr = 0;
    let output = 0;

    for (let each of timeSeries){
        if (each > curr)
        output += duration;
        else
        output += each + duration - curr;

        curr = each + duration;
    }
    return output;
};